roles/installtools/tasks/main.yml

```diff
@@ -14,66 +14,65 @@
     name: "{{ bioblend_packages }}"
     state: latest
 
-- name: Create tool list directory on the server
+- name: Create new empty tool list directory on the server
   file:
     path: "{{work_dir}}"
-    state: directory
+    state: "{{item}}"
+  with_items:
+    - absent
+    - directory
 
-- name: Check whether working directory is empty
+- name: Find yml files in tool list directory
   find:
-    paths: "{{work_dir}}"
+    paths: "{{tool_list_dir}}"
     patterns:
-      - "*"
-  register: work_dir_files
-
-- name: show work_dir_files
-  debug:
-    msg: "{{work_dir_files.matched}}"
-    verbosity: 1
-
-- name: Fail if directory is not empty
-  fail:
-    msg: "Directory already contains files with .yml or .yaml extension or contains an old shed_install.py. Please remove these files or specify another directory. {{work_dir_files.files}}"
-  when: '"{{work_dir_files.matched}}" > "0"'
-
-- name: Move tool directory on galaxy server
-  copy:
-    dest: "{{work_dir}}"
-    src: "{{tool_list_dir}}"
-    force: yes
-
-- name: Create a file list from the directory
-  find: 
-    paths: "{{work_dir}}"
-    patterns: 
       - "*.yml"
       - "*.yaml"
   register: tools
+  delegate_to: 127.0.0.1
+  become: no
 
-- name: Set list of tool list files as a fact
-  set_fact:
-    tool_list_files: "{{tools|json_query('files[*].path')}}"
+- name: Show which yamls will be transferred
+  debug: 
+    var: tools
+    verbosity: 1
 
 - name: delete api key from tool list files
   lineinfile:
-    destfile: "{{item}}"
+    destfile: "{{item.path}}"
     regexp: "api_key:.*"
     state: absent
-#    line: "api_key{{':'}} {{galaxy_master_api_key}}"
-  with_items: "{{tool_list_files}}"
-
-- name: Show tool_list_files
-  debug:
-    var: tool_list_files
-    verbosity: 1
+  with_items: 
```

Comment:
This item is a list. More items are possible if more items are specified by the user.